host/linux/aesm.c

```diff
@@ -217,7 +228,7 @@ static oe_result_t _unpack_var_int(
     uint8_t tag;
     uint8_t tmp_tag;
 
-    if ((*pos = _unpack_tag(buf, *pos, &tag)) == -1)
+    if ((*pos = (size_t)_unpack_tag(buf, *pos, &tag)) == OE_ERROR_UNPACK)
```

Comment:
Huh, this whole situation feels really awkward (not your code changes, but the way the file currently is).

The function "_unpack_tag" is defined in this file and used only within this file, but every time this function gets called we cast it from signed to unsigned.  And then we assign it to an unsigned value, which is then compared to a signed value cast into an unsigned value.  Previously these warnings were hidden, but with the new warning flags you're enabling, unfortunately you have to deal with this 😇 

I think it makes sense to return a signed type for the _unpack_tag function, which follows the normal convention of "negative returns are errors". But maybe we can rewrite this line to be something like:

```
ssize_t pos_new = _unpack_tag(buf, *pos, &tag);
if (pos_new == -1)
    OE_RAISE(OE_FAILURE);
*pos = (size_t)pos_new;
```

Hmm.. But you know, this way I've suggested of doing it is pretty ugly (and 4 lines instead of 2 lines). The lines that you have submitted here I think are more concise and arguably more easy to read. So honestly, I don't know what the best way to simplify this is.  What do you think? I'm on the fence here

Reply:
(This also applies to all other calls that do something similar here in this file)

Reply:
As part of PR #973, I had first changed the type of `pos` to `ssize_t` altogether. But it required further casts during arithemetics inside the `_unpack_tag` function. So I modified this as per Simon's input.